src/test_util/mod.rs

```diff
@@ -309,22 +309,32 @@ where
         .unwrap()
 }
 
-pub async fn wait_for<F, Fut>(mut f: F)
+pub async fn wait_for<F, Fut>(mut f: F, wait_for_secs: u64)
```

Comment:
Updated this to take a `wait_for_secs`, to make this slightly more configurable as I was diagnosing a race condition. The default of 5 seconds turned out to be okay, but I left this in to give that extra control to the caller.

Reply:
Maybe more flexible [std::time::Duration](https://doc.rust-lang.org/std/time/struct.Duration.html) instead secsonds?

Reply:
Good idea!